### PR TITLE
Change unusual_byte_groupings to require byte groupings of equal size

### PR DESCRIPTION
**Pull request by nindalf**
Fixes issue #6556

This lint required byte groupings of size 2 or 4 for `Radix::Binary` and `Radix::Hexadecimal`. Since there are good reasons for allowing groups of other sizes, this PR relaxes the restriction. This lint now requires that 

- group sizes after the first group be of the same size and
- greater or equal in size to the first group.

---

changelog: [`unusual_byte_groupings`]: reduce false positives by relaxing restriction requiring groups of specific sizes.

### CODE REVIEW

File: clippy_lints/src/literal_representation.rs [hunk 1]
```diff
@@ -210,7 +210,7 @@ impl WarningType {
                 cx,
                 UNUSUAL_BYTE_GROUPINGS,
                 span,
-                "digits of hex or binary literal not grouped by four",
+                "digits of hex, binary or octal literal not groups of equal size",
```
Comment (llogiq) [inline clippy_lints/src/literal_representation.rs:213]:
```suggestion
                "digits of hex, binary or octal literal not in groups of equal size",
```

File: clippy_lints/src/literal_representation.rs [hunk 2]
```diff
@@ -427,8 +427,12 @@ impl LiteralDigitGrouping {
 
         let first = groups.next().expect("At least one group");
 
-        if (radix == Radix::Binary || radix == Radix::Hexadecimal) && groups.any(|i| i != 4 && i != 2) {
-            return Err(WarningType::UnusualByteGroupings);
+        if radix == Radix::Binary || radix == Radix::Octal || radix == Radix::Hexadecimal {
```
Comment (llogiq) [inline clippy_lints/src/literal_representation.rs:430]:
```suggestion
        if radix != Radix::Decimal {
```

Comment (nindalf) [reply clippy_lints/src/literal_representation.rs:430]: I think this keeps the intention clearer.

**Comment by llogiq** [comment]
Thank you! @bors r+

**Pull request merged**